thrust/system/cuda/memory_resource.h

```diff
@@ -82,11 +83,123 @@ namespace detail
         return ::cudaMallocManaged(ptr, bytes, cudaMemAttachGlobal);
     }
 
+    /*! A version of thrust::cuda_cub::pointer that uses c++ references instead
+     * of thrust::cuda::reference. This is to allow managed memory pointers to
+     * be used with host-side code in standard libraries that are not compatible
+     * with proxy references.
+     */
+    template <typename T>
+    class managed_pointer
```

Comment:
Why is this class here and not in pointer.h?

Reply:
It seemed more like an implementation detail of `managed_memory_resource`, so I left it here. It seems like it could fit either place to me.